f5_os_test/testrunner_containers/docker_manager.py

```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env python
+# Copyright 2015-2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+'''library to create and use containers an environment with a local registry
+and a container orchestration environment'''
+
+import jinja2
+import logging
+import os
+from os.path import join
+import subprocess
+import sys
+import time
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+CTXT = os.path.abspath(os.curdir)
+REG_ID = 'docker-registry.pdbld.f5net.com/f5-openstack-test'
+TIMESTAMP = time.time()
+def render_dockerfile(**kwargs):
+    infname = join(kwargs['test_type'], 'project_docker.tmpl')
+    outdir = join(kwargs['test_type'], kwargs['project'])
+    outfname = join(outdir, 'Dockerfile')
+    logging.debug('outfname: {}'.format(outfname))
+    open(outfname, 'w').write(
+        jinja2.Template(open(infname).read()).render(**kwargs)
+    )
+
+def _publish_testrunner_container(registry_fullname):
+    pubstring = "docker push {}".format(registry_fullname)
```

Comment:
Think about using docker-py (or some other Python lib for docker) in the future. Might be cool to checkout this:
https://docker-py.readthedocs.io/en/latest/api/